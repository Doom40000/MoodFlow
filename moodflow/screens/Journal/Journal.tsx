import React, { useState, useEffect } from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import {
  FlatList,
  Text,
  View,
  Modal,
  Pressable,
  TextInput,
} from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

import styles from './styles';
import JournalListItem from '../../components/JournalListItem/JournalListItem';
import JournalEntry from '../JournalEntry/JournalEntry';
import {
  getAllJournalsController,
  createJournalEntryController,
} from '../../native-db/controllers/journal';

const Stack = createStackNavigator();

interface Journal {
  body: string;
  created_at: string;
  id: number;
  title: string;
}

const Journal = () => {
  const [journalEntries, setJournalEntries] = useState<any>([]);
  const [modalVisible, setModalVisible] = useState<boolean>(false);
  const [entryTitle, setEntryTitle] = useState<string>('');
  const [entryBody, setEntryBody] = useState<string>('');

  useEffect(() => {
    fetchJournals();
  }, []);

  const fetchJournals = async () => {
    const result = await getAllJournalsController();
    setJournalEntries(result);
  };

  const saveEntry = async () => {
    if (!entryBody || !entryTitle) return;
    await createJournalEntryController(entryTitle, entryBody);
    await fetchJournals();
    setEntryTitle('');
    setEntryBody('');
    setModalVisible(!modalVisible);
  };

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.headerContainer}>
        <Text style={styles.bigHeading}>Your Journal</Text>
        <Pressable
          onPress={() => setModalVisible(true)}
          style={styles.noteButton}
        >
          <Text style={styles.buttonTextPrimary}>+</Text>
        </Pressable>
      </View>
      <Modal
        animationType="slide"
        visible={modalVisible}
        onRequestClose={() => {
          setModalVisible(!modalVisible);
        }}
      >
        <View style={styles.modalContainer}>
          <View>
            <Text style={styles.bigHeading}>Add journal entry ✍️</Text>
          </View>
          <View>
            <Text style={styles.mediumHeading}>Add a title</Text>
            <TextInput
              style={styles.smallInput}
              onChangeText={(text) => setEntryTitle(text)}
              defaultValue={entryTitle}
            />
          </View>
          <View>
            <Text style={styles.mediumHeading}>Add your thoughts</Text>
            <TextInput
              editable
              multiline
              style={styles.bigInput}
              onChangeText={(text) => setEntryBody(text)}
              defaultValue={entryBody}
            />
          </View>
          <View>
            <Pressable onPress={() => saveEntry()} style={styles.buttonPrimary}>
              <Text style={styles.buttonTextPrimary}>Save thought</Text>
            </Pressable>
            <Pressable
              onPress={() => setModalVisible(!modalVisible)}
              style={styles.buttonSecondary}
            >
              <Text style={styles.buttonTextSecondary}>Close</Text>
            </Pressable>
          </View>
        </View>
      </Modal>
      {journalEntries ? (
        <FlatList
          data={journalEntries}
          renderItem={({ item }) => <JournalListItem journalItem={item} />}
        />
      ) : (
        <Text>No journal entries</Text>
      )}
    </SafeAreaView>
  );
};

export default Journal;
