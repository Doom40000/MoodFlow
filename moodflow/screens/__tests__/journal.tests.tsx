import { render } from '@testing-library/react-native';
import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

import Journal from '../Journal/Journal';
import MockedNavigator from '../../components/MockNavigationForTests/MockNavigation';

jest.useFakeTimers();

describe(Journal, () => {
  it('renders correctly', () => {
    const tree = renderer.create(<MockedNavigator component={Journal} params={{}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});
