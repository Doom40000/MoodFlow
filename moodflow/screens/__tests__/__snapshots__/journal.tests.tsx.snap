// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Journal renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              />
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  MockedScreen
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          pointerEvents="box-none"
          sheetAllowedDetents="large"
          sheetCornerRadius={-1}
          sheetExpandsWhenScrolledToEdge={true}
          sheetGrabberVisible={false}
          sheetLargestUndimmedDetent="all"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              [
                {
                  "display": "flex",
                  "overflow": undefined,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              {
                "close": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  {
                    "flex": 1,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    [
                      {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      [
                        {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <RNCSafeAreaView
                        edges={
                          {
                            "bottom": "additive",
                            "left": "additive",
                            "right": "additive",
                            "top": "additive",
                          }
                        }
                        style={
                          {
                            "alignContent": "center",
                            "backgroundColor": "#1FADD7",
                            "flex": 1,
                            "width": "100%",
                          }
                        }
                      >
                        <View
                          style={
                            {
                              "flex": 0.6,
                            }
                          }
                        >
                          <RNSVGSvgView
                            bbHeight="1334"
                            bbWidth="750"
                            focusable={false}
                            height={1334}
                            style={
                              [
                                {
                                  "backgroundColor": "transparent",
                                  "borderWidth": 0,
                                },
                                {
                                  "flex": 0,
                                  "height": 1334,
                                  "width": 750,
                                },
                              ]
                            }
                            width={750}
                          >
                            <RNSVGGroup
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                            >
                              <RNSVGImage
                                align="xMidYMin"
                                fill={
                                  {
                                    "payload": 4278190080,
                                    "type": 0,
                                  }
                                }
                                height="1334"
                                meetOrSlice={0}
                                src={null}
                                testID="Logo"
                                width="750"
                                x="0"
                                y="111.16666666666667"
                              />
                            </RNSVGGroup>
                          </RNSVGSvgView>
                        </View>
                        <View
                          style={
                            {
                              "paddingBottom": 20,
                              "paddingHorizontal": 20,
                            }
                          }
                        >
                          <View
                            style={
                              {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                              }
                            }
                          >
                            <Text
                              style={
                                {
                                  "alignSelf": "center",
                                  "color": "#FFFFFF",
                                  "fontSize": 32,
                                  "marginBottom": 24,
                                }
                              }
                            >
                              Your Journal
                            </Text>
                            <View
                              accessibilityState={
                                {
                                  "busy": undefined,
                                  "checked": undefined,
                                  "disabled": undefined,
                                  "expanded": undefined,
                                  "selected": undefined,
                                }
                              }
                              accessibilityValue={
                                {
                                  "max": undefined,
                                  "min": undefined,
                                  "now": undefined,
                                  "text": undefined,
                                }
                              }
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onBlur={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                {
                                  "alignSelf": "center",
                                  "backgroundColor": "#FFFFFF",
                                  "borderRadius": 4,
                                  "height": 40,
                                  "width": 40,
                                }
                              }
                            >
                              <Text
                                style={
                                  {
                                    "alignSelf": "center",
                                    "color": "#1FADD7",
                                    "fontSize": 18,
                                  }
                                }
                              >
                                +
                              </Text>
                            </View>
                          </View>
                          <Modal
                            animationType="slide"
                            hardwareAccelerated={false}
                            onRequestClose={[Function]}
                            visible={false}
                          />
                          <RCTScrollView
                            data={[]}
                            getItem={[Function]}
                            getItemCount={[Function]}
                            keyExtractor={[Function]}
                            onContentSizeChange={[Function]}
                            onLayout={[Function]}
                            onMomentumScrollBegin={[Function]}
                            onMomentumScrollEnd={[Function]}
                            onScroll={[Function]}
                            onScrollBeginDrag={[Function]}
                            onScrollEndDrag={[Function]}
                            removeClippedSubviews={false}
                            renderItem={[Function]}
                            scrollEventThrottle={50}
                            stickyHeaderIndices={[]}
                            viewabilityConfigCallbackPairs={[]}
                          >
                            <View />
                          </RCTScrollView>
                        </View>
                      </RNCSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
